server.py

```diff
@@ -147,13 +147,17 @@ def do_save_response():
 
 
 @app.route('/responses', methods=['GET'])
-def do_get_responses():
+@app.route('/responses/<invalid>', methods=['GET'])
```

Comment:
This route clashes with `/responses/<mongo_id>` below

A separate route on `/invalid-responses` to get all invalid (and maybe `/invalid-responses/<mongo_id>` for a specific one) would work better?